### ISSUE TITLE
`az resource list -o table --query "[].{name:name, location:location, type:type}"` not displaying the type

### ISSUE THREAD

**Issue by brupelo** [Azure/azure-cli#24520]
If you run `az resource list -o table --query "[].{name:name, location:location, type:type}"` it won't display the type column while
if you do `az resource list -o table --query "[].{name:name, location:location, kind:type}"` will be shown.

I guess it's because it's conflicting somehow with the reserved python's keyword `type`? Is this a bug?

Btw, I don't see in the help any way to sort this table by column... is it possible to achieve somehow?

**Comment by RakeshMohanMSFT** [comment]
@brupelo This will give you the type `az resource list -o table --query "[].type"`

**Comment by brupelo** [comment]
@RakeshMohanMSFT  Yeah, I know and you're correct but please read again my question, I'm not trying to display only the "unnamed type" but I'm trying to display 3 columns, each one with custom name and if you run

`az resource list -o table --query "[].{name:name, location:location, type:type}"` you won't get the `type` column shown.

But if you run instead:

`az resource list -o table --query "[].{name:name, location:location, kind:type}"`

Look how I've used `kind` as the column name it will be displayed... my guess it's because there is a bug here preventing to use reserved keywords from python as column labels.

Also, it should be easy to sort out columns in the `table` format but currently I haven't seen that option... The only workaround is using `tsv`, which it's not very convenient

**Comment by yonzhan** [comment]
route to CXP team

**Comment by RakeshMohanMSFT** [comment]
@brupelo Apologies for misunderstanding the question. Let me send it to the service team for further troubleshooting.
